src/compressed_tensors/logger.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Logger configuration for Compressed Tensors.
+"""
+
+import os
+import sys
+from dataclasses import dataclass
+from typing import Any, Dict, Optional
+
+from loguru import logger
+
+
+__all__ = ["LoggerConfig", "configure_logger", "logger"]
+
+
+# used by `support_log_once``
+_logged_once = set()
+
+
+@dataclass
+class LoggerConfig:
+    disabled: bool = False
+    clear_loggers: bool = True
+    console_log_level: Optional[str] = "INFO"
+    log_file: Optional[str] = None
+    log_file_level: Optional[str] = None
+
+
+def configure_logger(config: Optional[LoggerConfig] = None) -> None:
+    """
+    Configure the logger for Compressed Tensors.
+    This function sets up the console and file logging
+    as per the specified or default parameters.
+
+    Note: Environment variables take precedence over the function parameters.
+
+    :param config: The configuration for the logger to use.
+    :type config: LoggerConfig
+    """
+    logger_config = config or LoggerConfig()
+
+    # env vars get priority
+    if (disabled := os.getenv("COMPRESSED_TENSORS_LOG_DISABLED")) is not None:
+        logger_config.disabled = disabled.lower() == "true"
+    if (clear_loggers := os.getenv("COMPRESSED_TENSORS_CLEAR_LOGGERS")) is not None:
+        logger_config.clear_loggers = clear_loggers.lower() == "true"
+    if (console_log_level := os.getenv("COMPRESSED_TENSORS_LOG_LEVEL")) is not None:
+        logger_config.console_log_level = console_log_level.upper()
+    if (log_file := os.getenv("COMPRESSED_TENSORS_LOG_FILE")) is not None:
+        logger_config.log_file = log_file
+    if (log_file_level := os.getenv("COMPRESSED_TENSORS_LOG_FILE_LEVEL")) is not None:
+        logger_config.log_file_level = log_file_level.upper()
```

Comment:
It's common for people to set env variables to `1` when triggering them (i.e. `COMPRESSED_TENSORS_LOG_DISABLED=1 ...`.) By explicitly comparing to `"true"`, we lose that ability.

Instead this could be 
```python
logger_config.disabled = bool(os.getenv("COMPRESSED_TENSORS_LOG_DISABLED"), logger_config.disabled))
```
Same for the clear loggers check.

Reply:
is there a lot of precedent for this already? if not, this makes sense, but consistency is more important

Reply:
I've definitely seen this before. e.g. in pytorch docs:

>PYTORCH_NO_CUDA_MEMORY_CACHING
 If set to 1, disables caching of memory allocations in CUDA. This can be useful for debugging.

https://docs.pytorch.org/docs/stable/cuda_environment_variables.html

Reply:
I mean precedence in terms of our code base env vars. If we can get by without them, all the better. if not and no precedent exists, allowing `1` makes sense to me

Reply:
I think we should move towards allowing `1` moving forward. Definitely good to move towards better patterns, if even if it's slightly different than what we did in the past